src/NHibernate/Type/Int32Type.cs

```diff
@@ -114,7 +115,7 @@ public override object DefaultValue
 
 		public override string ObjectToSQLString(object value, Dialect.Dialect dialect)
 		{
-			return value.ToString();
+			return Convert.ToInt32(value).ToString(CultureInfo.InvariantCulture);
```

Comment:
Can we do downcast instead? `return ((Int32)value).ToString(CultureInfo.InvariantCulture);`?

Reply:
https://github.com/nhibernate/nhibernate-core/pull/3547#discussion_r1632428617

> It could be a direct cast too, but that may break working cases where the value was already a valid string representation of the type.

For many types, the static field injection in queries features could work with non-matching types. The convert can already be a possible breaking change. The cast seems to me as possibly breaking more cases.

There are also a bunch of types who do cast to their expected type. So, for consistency, it will be better for all types to the same.

Still, here, we are in a patch PR, so, I tend to favor minimize the possibilities of introducing possible breaking changes for some cases. But the possibly affected cases are also corner uses of a corner case feature. So, why not.

Reply:
@fredericDelaporte can you explain why you've changed your opinion?

Reply:
Isn't what does my previous comment? In other words, I have no strong opinion on that subject. Both options have advantages and drawbacks.